server/src/main/java/org/elasticsearch/indices/flush/SyncedFlushService.java

```diff
@@ -412,8 +433,11 @@ private PreSyncedFlushResponse performPreSyncedFlush(PreShardSyncedFlushRequest
         FlushRequest flushRequest = new FlushRequest().force(false).waitIfOngoing(true);
         logger.trace("{} performing pre sync flush", request.shardId());
         Engine.CommitId commitId = indexShard.flush(flushRequest);
-        logger.trace("{} pre sync flush done. commit id {}", request.shardId(), commitId);
-        return new PreSyncedFlushResponse(commitId);
+        // If there is another commit between flush and get commit stats, the sync-flush will fail as commitId is mismatch,
+        // Therefore, it's safe to execute these two commands separately here.
+        final int numDocs = indexShard.commitStats().getNumDocs();
```

Comment:
can we check the commit id in the stats and make sure it's the same? alternatively, can we maybe find the segments info for the right commit and create a stats object there? You're right about the synced flush not caring, but  I'm worried that we get false scary warn message. I want to avoid that.